### PR TITLE
Add deletion protection and access logs from ALB (using same approach)

### PR DESCRIPTION
**Pull request by dunnill**
##### Corresponding Issue(s):
 - PRs should have a corresponding issue. If no issue exists, provide details in the **Reason for Change(s)** section

##### Summary of change(s):
Bring in resources and variables from alb pertaining to access logs and deletion protection.

##### Reason for Change(s):
Customer has requested deletion protection and access logs to be enabled on NLB.  We're advising them of the limitations of using access logs - i.e. TLS.  

- If a bug, describe error scenario, including expected behavior and actual behavior.

- If an enhancement, describe the use case and the perceived benefit(s).
Allow customers to protect NLB from deletion and turn on access logs for TLS information.

##### Will the change trigger resource destruction or replacement? If yes, please provide justification:
No `create_logging_bucket` and `enable_deletion_protection` default to false so no changes will occur by default

##### Does this update/change involve issues with other external modules? If so, please describe the scenario.
No
##### If input variables or output variables have changed or has been added, have you updated the README?
Yes
##### Do examples need to be updated based on changes?
No
##### Note to the PR requester about Closing PR's
Please message the person that opened the issue when auto closing it on slack, as well as any other stake holders of deep interest. Only close the issue if you believe that the issue is fully resolved with this PR.

### CODE REVIEW

**Review by stevengorrell — DISMISSED** [review]

File: variables.tf [hunk 1]
```diff
@@ -4,6 +4,18 @@ variable "create_internal_zone_record" {
...
+  type        = bool
+  default     = false
+}
+
+variable "enable_deletion_protection" {
+  description = "If true, deletion of the load balancer will be disabled via the AWS API. This will prevent Terraform from deleting the load balancer. Defaults to false."
+  type        = bool
+  default     = false
+}
+
```
Comment (johnctitus) [inline variables.tf:18]: It's nitpicky, but we should move this variable between the `cross_zone` and `eni_count` variables, as the list should be alphabetized.

**Review by johnctitus — CHANGES_REQUESTED** [review]
Hey @dunnill thanks for the contribution.

It looks like the CI test is currently failing linting.  It looks like the readme is out of sync and need to be regenerated again.

I think we should also enable the logging along with the force destroy option on one of the two NLBs created for the tests.  Probably the external one, as it appears to have less going on.
https://github.com/rackspace-infrastructure-automation/aws-terraform-nlb/blob/5c777c2ee423e4c3054a540cd6c7f4602afca2d4/tests/test1/main.tf#L110-L133

Everything else looks good, assuming nothing crops up in the final CI job.

**Comment by dunnill** [comment]
Thanks @stevengorrell and @johnctitus for all you assistance.  Hopefully next time it'll go green sooner.  Now it is working and tested - are you okay to 👍 the PR so I can merge.  Thanks

**Review by johnctitus — APPROVED** [review]
LGTM.  Thanks again for the contribution @dunnill

**Review by stevengorrell — APPROVED** [review]

**Pull request merged**